packages/compat/src/fixup-rules.js

```diff
@@ -0,0 +1,256 @@
+	["getFirstToken", "getFirstToken"],
+	["getFirstTokens", "getFirstTokens"],
+	["getLastToken", "getLastToken"],
+	["getLastTokens", "getLastTokens"],
+	["getTokenAfter", "getTokenAfter"],
+	["getTokenBefore", "getTokenBefore"],
+	["getTokenByRangeStart", "getTokenByRangeStart"],
+	["getTokens", "getTokens"],
+	["getTokensAfter", "getTokensAfter"],
+	["getTokensBefore", "getTokensBefore"],
+	["getTokensBetween", "getTokensBetween"],
+]);
+
+/**
+ * Tracks the original rule definition and the fixed-up rule definition.
+ * @type {WeakMap<FixupRuleDefinition|FixupLegacyRuleDefinition,FixupRuleDefinition>}
+ */
+const fixedUpRuleReplacements = new WeakMap();
+
+/**
+ * Tracks all of the fixed up rule definitions so we don't duplicate effort.
+ * @type {WeakSet<FixupRuleDefinition>}
+ */
+const fixedUpRules = new WeakSet();
+
+/**
+ * Tracks the original plugin definition and the fixed-up plugin definition.
+ * @type {WeakMap<FixupPluginDefinition,FixupPluginDefinition>}
+ */
+const fixedUpPluginReplacements = new WeakMap();
+
+/**
+ * Tracks all of the fixed up plugin definitions so we don't duplicate effort.
+ * @type {WeakSet<FixupPluginDefinition>}
+ */
+const fixedUpPlugins = new WeakSet();
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * Takes the given rule and creates a new rule with the `create()` method wrapped
+ * to provide the missing methods on the `context` object.
+ * @param {FixupRuleDefinition|FixupLegacyRuleDefinition} ruleDefinition The rule to fix up.
+ * @returns {FixupRuleDefinition} The fixed-up rule.
+ */
+export function fixupRule(ruleDefinition) {
+	// first check if we've already fixed up this rule
+	if (fixedUpRuleReplacements.has(ruleDefinition)) {
+		return fixedUpRuleReplacements.get(ruleDefinition);
+	}
+
+	const isLegacyRule = typeof ruleDefinition === "function";
+
+	// check to see if this rule definition has already been fixed up
+	if (!isLegacyRule && fixedUpRules.has(ruleDefinition)) {
+		return ruleDefinition;
+	}
+
+	const originalCreate = isLegacyRule
+		? ruleDefinition
+		: ruleDefinition.create.bind(ruleDefinition);
+
+	const ruleCreate = context => {
+		// if getScope is already there then no need to create old methods
+		if ("getScope" in context) {
+			return originalCreate(context);
+		}
+
+		const sourceCode = context.sourceCode;
+		let currentNode = sourceCode.ast;
+
+		const newContext = Object.assign(Object.create(context), {
+			parserServices: sourceCode.parserServices,
+
+			/*
+			 * The following methods rely on the current node in the traversal,
+			 * so we need to add them manually.
+			 */
+			getScope() {
+				return sourceCode.getScope(currentNode);
+			},
+
+			getAncestors() {
+				return sourceCode.getAncestors(currentNode);
+			},
+
+			getDeclaredVariables() {
+				return sourceCode.getDeclaredVariables(currentNode);
+			},
```

Comment:
`context.getDeclaredVariables` wasn't related to the current node, it had a `node` parameter.

Reply:
Oops, good catch. 👍